webservices/legal_docs/advisory_opinions.py

```diff
@@ -286,26 +315,110 @@ def parse_ao_citations(text, ao_component_to_name_map):
             new_title, new_section = reclassify_statutory_citation(
                 citation.group('title'), citation.group('section'))
             matches.add((
                 int(new_title),
                 str(new_section)
             ))
+        for possible_multiple_citation in MULTIPLE_STATUTE_CITATION_REGEX.finditer(text):
+            citations_title = possible_multiple_citation.group('title')
+            possible_sections = possible_multiple_citation.group('possible_sections')
+
+            for section in STATUTE_SECTION_ONLY_REGEX.finditer(possible_sections):
+                new_title, new_section = reclassify_statutory_citation(
+                    citations_title, section.group('section'))
+
+                if validate_statute_citation(new_title, new_section):
+                    matches.add((
+                        int(new_title),
+                        str(new_section)
+                    ))
+                else:
+                    logger.debug("Citation out of range - excluding {} USC {}".format(new_title, new_section))
     return matches
 
 
+def validate_regulation_citation(part):
+    """
+    Check to see if the regulation (CFR) is in the expected range. Add padding for future regs.
+
+    Source: https://www.gpo.gov/fdsys/pkg/CFR-2018-title11-vol1/pdf/CFR-2018-title11-vol1-chapI.pdf
+
+    11 CFR: 1-8, 100-120, 200-205, 300-305, 9001-9099
+
+    Because we're only checking 11 CFR, fall back to False and drop anything out of range
+    """
+    try:
+        part = int(part)
+    except:
+        return False
+
+    return (1 <= part <= 8 or
+        100 <= part <= 120 or
+        200 <= part <= 205 or
+        300 <= part <= 305 or
+        9001 <= part <= 9099)
+
+
 def parse_regulatory_citations(text):
     matches = set()
     if text:
-        for citation in REGULATION_CITATION_REGEX.finditer(text):
+        for citation in SINGLE_REGULATION_CITATION_REGEX.finditer(text):
             matches.add((
                 int(citation.group('title')),
                 int(citation.group('part')),
                 int(citation.group('section'))
             ))
+
+        for possible_multiple_citation in MULTIPLE_REGULATION_CITATION_REGEX.finditer(text):
+            citations_title = possible_multiple_citation.group('title')
+            possible_parts_and_sections = possible_multiple_citation.group('possible_parts_and_sections')
+
+            for part_and_section in REGULATION_SECTION_ONLY_REGEX.finditer(possible_parts_and_sections):
+
+                if validate_regulation_citation(part_and_section.group('part')):
+                    matches.add((
+                        int(citations_title),
+                        int(part_and_section.group('part')),
+                        int(part_and_section.group('section'))
+                    ))
+                else:
+                    logger.debug("Citation out of range - excluding {} CFR {}.{}".
```

Comment:
log level should be INFO

Reply:
@pkfec these are at the `debug` level because we're processing so much. On the legal side, only essential logs are on the `info` level.